File: py/features.py
# AppStoreMetadata SDK feature factory

from feature.base_feature import AppStoreMetadataBaseFeature
from feature.test_feature import AppStoreMetadataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AppStoreMetadataBaseFeature(),
        "test": lambda: AppStoreMetadataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()

File: rb/features.rb
# AppStoreMetadata SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module AppStoreMetadataFeatures
  def self.make_feature(name)
    case name
    when "base"
      AppStoreMetadataBaseFeature.new
    when "ratelimit"
      AppStoreMetadataRatelimitFeature.new
    when "retry"
      AppStoreMetadataRetryFeature.new
    when "test"
      AppStoreMetadataTestFeature.new
    when "timeout"
      AppStoreMetadataTimeoutFeature.new
    else
      AppStoreMetadataBaseFeature.new
    end
  end
end
